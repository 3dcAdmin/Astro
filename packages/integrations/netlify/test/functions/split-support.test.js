import { expect } from 'chai';
import netlifyAdapter from '../../dist/index.js';
import { loadFixture, testIntegration } from './test-utils.js';

describe('Split support', () => {
	/** @type {import('./test-utils').Fixture} */
	let fixture;

	before(async () => {
		fixture = await loadFixture({
			root: new URL('./fixtures/split-support/', import.meta.url).toString(),
			output: 'server',
			adapter: netlifyAdapter({
				dist: new URL('./fixtures/split-support/dist/', import.meta.url),
			}),
			site: `http://example.com`,
			integrations: [testIntegration()],
			build: {
				split: true,
			},
		});
		await fixture.build();
	});

	it('outputs a correct redirect file', async () => {
		const redir = await fixture.readFile('/_redirects');
		const blogRouteIndex = redir.indexOf(
			'/blog    /.netlify/functions/src/pages/entry.blog.astro     200'
		);
		const baseRouteIndex = redir.indexOf(
			'/        /.netlify/functions/src/pages/entry.index.astro    200'
		);

		expect(baseRouteIndex).to.not.be.equal(-1);
		expect(blogRouteIndex).to.not.be.equal(-1);
		expect(redir).to.matchSnapshot();
	});

	describe('Should create multiple functions', () => {
		it('for index page', async () => {
			const entryURL = new URL(
				'./fixtures/split-support/.netlify/functions-internal/src/pages/entry.index.astro.mjs',
				import.meta.url
			);
			console.debug(entryURL);
			console.debug(entryURL.toString());
			const { handler } = await import(entryURL);
			const resp = await handler({
				httpMethod: 'POST',
				headers: {},
				rawUrl: 'http://example.com/',
				body: '{}',
			});
			expect(resp.statusCode).to.equal(200);
		});
		it('for blog page', async () => {
			const { handler } = await import(
				'./fixtures/split-support/.netlify/functions-internal/src/pages/entry.blog.astro.mjs'
			);
			const resp = await handler({
				httpMethod: 'POST',
				headers: {},
				rawUrl: 'http://example.com/blog',
				body: '{}',
			});
			expect(resp.statusCode).to.equal(200);
		});
	});
});
