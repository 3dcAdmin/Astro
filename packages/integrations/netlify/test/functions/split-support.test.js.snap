// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Split support outputs a correct redirect file 1`] = `
"/blog    /.netlify/functions/src/pages/entry.blog.astro     200
/        /.netlify/functions/src/pages/entry.index.astro    200"
`;
